libraries/chain/include/eosio/chain/webassembly/interface.hpp

```diff
@@ -315,61 +315,61 @@ namespace eosio { namespace chain { namespace webassembly {
 
          REGISTER_LEGACY_HOST_FUNCTION(db_store_i64);
          REGISTER_LEGACY_HOST_FUNCTION(db_update_i64);
-         REGISTER_LEGACY_HOST_FUNCTION(db_remove_i64);
+         REGISTER_HOST_FUNCTION(db_remove_i64);
          REGISTER_LEGACY_HOST_FUNCTION(db_get_i64);
-         REGISTER_HOST_FUNCTION(db_next_i64);
+         REGISTER_LEGACY_HOST_FUNCTION(db_next_i64);
          REGISTER_LEGACY_HOST_FUNCTION(db_previous_i64);
-         REGISTER_LEGACY_HOST_FUNCTION(db_find_i64);
-         REGISTER_LEGACY_HOST_FUNCTION(db_lowerbound_i64);
-         REGISTER_LEGACY_HOST_FUNCTION(db_upperbound_i64);
-         REGISTER_LEGACY_HOST_FUNCTION(db_end_i64);
+         REGISTER_HOST_FUNCTION(db_find_i64);
+         REGISTER_HOST_FUNCTION(db_lowerbound_i64);
+         REGISTER_HOST_FUNCTION(db_upperbound_i64);
+         REGISTER_HOST_FUNCTION(db_end_i64);
 
 
          // uint64_t secondary index api
-         int32_t db_idx64_store(uint64_t, uint64_t, uint64_t, uint64_t, const uint64_t&);
-         void db_idx64_update(int32_t, uint64_t, const uint64_t&);
+         int32_t db_idx64_store(uint64_t, uint64_t, uint64_t, uint64_t, legacy_ptr<const uint64_t>);
+         void db_idx64_update(int32_t, uint64_t, legacy_ptr<const uint64_t>);
          void db_idx64_remove(int32_t);
          int32_t db_idx64_find_secondary(uint64_t, uint64_t, uint64_t, legacy_ptr<const uint64_t>, legacy_ptr<uint64_t>);
-         int32_t db_idx64_find_primary(uint64_t, uint64_t, uint64_t, uint64_t&, uint64_t);
+         int32_t db_idx64_find_primary(uint64_t, uint64_t, uint64_t, legacy_ptr<uint64_t>, uint64_t);
          int32_t db_idx64_lowerbound(uint64_t, uint64_t, uint64_t, legacy_ptr<uint64_t, 8>, legacy_ptr<uint64_t, 8>);
          int32_t db_idx64_upperbound(uint64_t, uint64_t, uint64_t, legacy_ptr<uint64_t, 8>, legacy_ptr<uint64_t, 8>);
```

Comment:
Explicit alignments are redundant.

Reply:
I made it explicit to avoid depending on implementation defined behavior.  I only used explicit alignment for the intrinsics where the alignment is observable from wasm.